packages/crawler/src/client.ts

```diff
@@ -1,115 +1,71 @@
-        }
-        timetables.map(async (timetable) => {
-            const hash = createHash('sha512').update(JSON.stringify(timetable.data.htmls)).digest('hex');
-            const timetableVersion = await dbClient<OptivumTimetableVersionsTable, { unqiue_id: number }>(
-                'optivum_timetable_versions',
-            )
-                .returning('unique_id')
-                .insert({
-                    school_rspo_id: school.rspo_id,
-                    generated_on: timetable.data.data.generationDate,
-                    timetable_data: JSON.stringify(timetable.data.data),
-                    discriminant: dbClient.raw(
-                        `(SELECT coalesce(max(discriminant), -1) + 1 FROM optivum_timetable_versions WHERE school_rspo_id = ${school.rspo_id} AND generated_on = '${timetable.data.data.generationDate}')`,
-                    ),
-                    hash,
-                })
-                .onConflict(['school_rspo_id', 'hash'])
-                .merge(['school_rspo_id'])
-                .returning('*')
-                .first();
-            await dbClient<TimetableUrlsTable, { id: number }[]>('timetable_urls')
-                .returning('id')
-                .insert({
-                    timetable_version_id: timetableVersion.unique_id,
-                    school_rspo_id: school.rspo_id,
-                    url: timetable.url,
-                })
-                .onConflict(['school_rspo_id', 'url', 'timetable_id'])
-                .merge(['last_check_at']);
-        });
-        console.log(`[RSPO ID: ${school.rspo_id}] Done!`);
-    });
+    const schools = await getSchoolsWithWebiste();
+    schools.map((school) => checkSchool(school, axiosInstance));
+}
+
+async function checkSchool(school: SchoolsTable & { website_url: string }, axiosInstance: Axios) {
+    const line = new SingleBar({ format: '[RSPO_ID: {rspo_id}] {status}', noTTYOutput: true });
+    line.start(1, 0, { status: 'Getting timetables...', rspo_id: school.rspo_id });
+    let optivumTimetables: string[];
+    try {
+        optivumTimetables = (await findTimetables(school.website_url, axiosInstance)).optivumTimetables;
+    } catch {
+        return;
+    }
+    if (optivumTimetables.length !== 0) {
+        line.update(0, { status: 'Parsing timetables and pushing data to database...' });
+    }
+    await Promise.allSettled(
+        optivumTimetables.map(async (timetableUrl) => {
+            const parsedTimetable = await parse(timetableUrl, axiosInstance);
+            const hash = createHash('sha512').update(JSON.stringify(parsedTimetable.htmls)).digest('hex');
+            const timetableVersion = await pushOptivumTimetableVersion(parsedTimetable, school.rspo_id, hash);
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            await pushOptivumTimetableVersionUrl(timetableUrl, timetableVersion!.unique_id, school.rspo_id);
```

Comment:
Czemu właściwie timetableVersion może być undefined/null? Jeśli pushOptivumTimetableVersion nigdy nie zwraca null, to w nim powinien być check, nie tutaj